src/ORM/DataObject.php

```diff
@@ -3064,13 +3064,31 @@ public function requireTable()
         $fields = $schema->databaseFields(static::class, false);
         $indexes = $schema->databaseIndexes(static::class, false);
         $extensions = self::database_extensions(static::class);
+        $legacyTables = $schema->getLegacyTableNames(static::class);
 
         if (empty($table)) {
             throw new LogicException(
                 "Class " . static::class . " not loaded by manifest, or no database table configured"
             );
         }
 
+        if ($legacyTables) {
+            $ignore = Config::inst()->get(static::class, 'ignored_legacy_tables') ?: [];
+            $renameTables = array_diff(
+                array_intersect($legacyTables, DB::table_list()),
+                $ignore
+            );
+            if (count($renameTables) > 1) {
+                throw new LogicException(
```

Comment:
Maybe just make this a warning and keep doing the update, otherwise users may end up with an inconsistent database.

Reply:
hmm, I guess the issue I have is if there is more than one legacy table - which one would be the better one to use?

Reply:
Don't do any rename; Create a brand new table in that case? It's equally likely that none of the tables are valid as is one of them is.